packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -477,8 +451,7 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
         style={{ width: offsetLeft }}
       />
       {cells}
-      {emptyCellWidth > 0 && <EmptyCell width={emptyCellWidth} />}
-      {rightCells.length > 0 && <div role="presentation" className={gridClasses.filler} />}
```

Comment:
Not sure why we need this + the empty cell. Perhaps I'm missing something for this.

Reply:
I don't remember :| You should build a test case with all the variations of:
 - pinned or no pinned columns
 - scrollbar or no scrollbar
 - empty space or no empty space (when columns don't fill the width)

And then check if all the cases behave correctly. Make sure to enable vertical borders on all cells, it will help you spot issues.

We should probably create a visual regression test based on that.